python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -756,35 +755,539 @@ def joinMatchListWithCatalog(self, matchCat, sourceCat):
         return joinMatchListWithCatalogImpl(self, matchCat, sourceCat)
 
 
-class ReferenceObjectLoader(ReferenceObjectLoaderBase):
-    """This class facilitates loading reference catalogs with gen 3 middleware.
+class ReferenceObjectLoader:
+    """This class facilitates loading reference catalogs.
 
     The QuantumGraph generation will create a list of datasets that may
     possibly overlap a given region. These datasets are then used to construct
-    and instance of this class. The class instance should then be passed into
+    an instance of this class. The class instance should then be passed into
     a task which needs reference catalogs. These tasks should then determine
     the exact region of the sky reference catalogs will be loaded for, and
     call a corresponding method to load the reference objects.
 
     Parameters
     ----------
     dataIds : iterable of `lsst.daf.butler.DataCoordinate`
-        An iterable object of data IDs that point to reference catalogs
-        in a gen 3 repository.
+        An iterable object of data IDs that point to reference catalogs.
     refCats : iterable of `lsst.daf.butler.DeferredDatasetHandle`
-        Handles to load refCats on demand
+        Handles to load refCats on demand.
     log : `lsst.log.Log`, `logging.Logger` or `None`, optional
         Logger object used to write out messages. If `None` a default
         logger will be used.
     """
+    ConfigClass = LoadReferenceObjectsConfig
+
     def __init__(self, dataIds, refCats, log=None, config=None, **kwargs):
+        if kwargs:
+            warnings.warn("Instantiating ReferenceObjectLoader with additional kwargs is deprecated "
+                          "and will be removed after v25.0", FutureWarning)
```

Comment:
Please use `stacklevel=2` here so that the warning appears to come from the caller code -- it makes it easier to find which of their lines had the errant kwarg. Generally stacklevel is always going to be useful.